opensfm/exif.py

```diff
@@ -57,7 +57,7 @@ def sensor_string(make, model):
     if make != 'unknown':
         # remove duplicate 'make' information in 'model'
         model = model.replace(make, '')
-    return (make.strip() + ' ' + model.strip()).lower()
+    return (make.strip() + ' ' + model.strip()).strip().lower()
```

Comment:
Ok, so let's assume make was not detected and it is empty string. Then we get extra space in the beginning of function result:
```
>>> make = ''
>>> model = 'model'
>>> print '{0:s} {1:s}'.format(make.strip(), model.strip()).lower()
 model
```
while with extra strip:
```
>>> print (make.strip() + ' ' + model.strip()).strip().lower()
model
```

It is related to xiaomi gimbal camera because that camera saves chinese(?) non utf8 valid bytes.

Reply:
> It is related to xiaomi gimbal camera because that camera saves chinese(?) non utf8 valid bytes.

Your concern is appreciated. However, we cannot and should not account for every possible violation of the EXIF specification (including every device firmware). The EXIF tags `Make` and `Model` must be ASCII only. Besides, devices exist which do not report anything, neither `Make` nor `Model`, and they work fine. So, it would be best to contact Xiaomi about a firmware update which would fix this particular issue.

Reply:
I would fully agree with you on that. But the world is not perfect and xiaomi support just sucks, that's why it would be nice to have at least something.

Reply:
Could you please share a link to an original image file, so that we could examine the EXIF data?

Reply:
https://imgur.com/a/Qcaghxf

Reply:
> https://imgur.com/a/Qcaghxf

Unfortunately, that site does not serve original images and strips image files of any headers. We would need an **original** or **unprocessed** file.

Reply:
Sorry, here is original image with all tags.

![img_20170101_010412](https://user-images.githubusercontent.com/150403/48287889-120cc980-e473-11e8-9893-72d2fcacc60e.jpg)

Reply:
So, I wanted to see what gives and tried a bunch of encodings on the `Make` EXIF tag. Things look like this:

Encoding | Decoded Text | Google Translation | Note
---------|--------------|--------------------|-----
UTF-16 | 놱뺩량쏗뿆벼폐쿞릫쮾 | 놱뺩 쏗 뿆 뿆 뿆 쿞 쿞 릫쮾 릫쮾 | Korean
Big5 | 控儔滄譙褪撮衄癹鼠侗 | Controlled sputum | non-sense
GB18030 | 北京飞米科技有限公司 | Beijing Feimi Technology Co., Ltd. | probable

Did you overwrite the `DateTime*` and `GPS*` EXIF tags? If not then the EXIF data is broken beyond repair.

Anyhow, we should probably store `"unknown YTXJ02FM": 6.17` in the database. It is not elegant but it works.

Reply:
No tags were overwritten. They all originally wrong/nonsense, like time, GPS position and etc.

BTW, correct company name is "Beijing FIMI Technology Co., Ltd"

Reply:
> No tags were overwritten. They all originally wrong/nonsense, like time, GPS position and etc.

Hmm, well if `DateTime` and GPS are broken *by default* then there is little to no need to bother adding this camera to the database, since OpenSfM cannot use this meta data anyway. In other words, if you really want to use imagery from this device with OpenSfM then you have to tag the imagery with correct and meaningful EXIF data beforehand. Consequently, while doing so you can also fix any invalid EXIF tags, like the `Make` tag.

Reply:
Well, yes, You are right and it's up to You whether merge it or not :)